server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/LongGaugeMetric.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.routing.allocation.allocator;
+
+import org.elasticsearch.telemetry.metric.LongGauge;
+import org.elasticsearch.telemetry.metric.LongWithAttributes;
+import org.elasticsearch.telemetry.metric.MeterRegistry;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * This wrapper allow us to record metric with APM (via LongGauge) while also access its current state via AtomicLong
+ */
+public record LongGaugeMetric(AtomicLong value, LongGauge gauge) {
```

Comment:
I wonder if we should move this to some utility folder, because it may be useful in other places in ES?

Reply:
Moved that into `org.elasticsearch.telemetry.metric`. @pgomulka, please let us know if there is a better place for that

Reply:
yes, I think this is good place. Thank you @idegtiarenko